monkey/infection_monkey/exploit/sshexec.py

```diff
@@ -170,6 +191,11 @@ def _exploit_host(self) -> ExploiterResultData:
                 ssh = self.exploit_with_login_creds(port)
             except FailedExploitationError:
                 self.exploit_result.error_message = "Exploiter SSHExploiter is giving up..."
+                self.publish_exploitation_event(
+                    target=self.host.ip_addr,
+                    exploitation_success=False,
+                    error_message=self.exploit_result.error_message,
+                )
                 logger.error(self.exploit_result.error_message)
                 return self.exploit_result
 
```

Comment:
Do we need to return a failed propagation event if the thread is interrupted?

Reply:
Break into its own PR. We can discuss there.